enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+- @bwplotka
+- @fpetkovski
+reviewers:
+- @alanmoran
+- @smarterclayton
+- @kbsingh
+- @pb82
+- @simonpasquier
+approvers:
+- TBD
+creation-date: 2021.08.12
+last-updated: 2021.08.12
+status: implementable
+---
+
+# Monitoring for Red Hat Managed Services
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [X] User-facing documentation is created in [openshift-docs]([https://github.com/openshift/openshift-docs/](https://github.com/openshift/openshift-docs/)) - (there are non required)
+
+## Summary
+
+This document proposes a solution and a delivery plan for scraping, querying and alerting on metrics related to Red Hat Managed Services running in OSD clusters. The proposed solution is not intended to complement the existing monitoring stack managed by the Cluster Monitoring Operator, nor used directly by Customers at this moment.
+
+### Glossary
+
+* MS-SRE/MT-SRE: Managed Service / Managed Tenant Site Reliability Engineer
+* MTO: Managed Tenant Owners - the teams that develop each Managed Service. The Kafka Team is one example of an MTO.
+* RHOBS: Red Hat Observability Service
+* RHODF: Red Hat Openshift Data Foundation
+
+For additional acronyms, see [Red Hat Dictionary](https://source.redhat.com/groups/public/red-hat-dictionary-or-lexicon/red_hat_abbreviations_acronyms_and_initialisms_dictionary_wiki)
+
+## Background
+
+Openshift Monitoring is currently composed of two main systems - Platform Monitoring (PM) and User Workload Monitoring (UWM), both of which are part of OpenShift core.
+
+Platform Monitoring holds metrics about core system components running in `openshift-*` and `kube-*` namespaces. Platform Monitoring is enabled in all OpenShift installations and cannot be disabled. Some of the components scraped by PM include core-dns, etcd, kube-apiserver, image-registry, operator-lifecycle-manager etc.
+Platform Monitoring is by design an immutable stack, [which means that admins cannot extend it with additional metrics without losing the support.](https://docs.openshift.com/container-platform/4.7/monitoring/configuring-the-monitoring-stack.html#support-considerations_configuring-the-monitoring-stack)
+
+User Workload Monitoring is an optional stack managed by Cluster Monitoring Operator (CMO) and is disabled by default. Once enabled by cluster admins, it allows users to monitor namespaces that are not part of OpenShift core.
+Please refer to the User Workload Monitoring [enhancement proposal](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/user-workload-monitoring.md) for more details. If enabled, only one stack can be deployed. The recommended way to get “platform” data is through Thanos Querier. Alternatively, Platform Monitoring Prometheus can also federate the latest data.
```

Comment:
Isn't it simply reference to possibility to federate data from Prometheus instance(s) living in Platform Monitoring stack? E.g. if I'm not mistaken that's what Telemeter client uses.

Reply:
Is it officially supported? AFAIK, the federation endpoint of PM Prometheus is exposed only for telemeter usage.

Reply:
FYI the RHOAM addon (and since RHMI) which has GA'd and had been operating in production for a while now is using platform prometheus federation to gather platform metrics and complement with middleware metrics for alerting.

Reply:
The RHODS addon is doing the same.